### PR TITLE
many: fix review comments from PR #3177

### PR DESCRIPTION
**Pull request by morphis**
Based on feedback from @niemeyer on https://github.com/snapcore/snapd/pull/3177

### CODE REVIEW

File: x11/xauth.go [hunk 1]
```diff
@@ -38,61 +38,48 @@ type xauth struct {
...
-
-func readChunk(f *os.File) ([]byte, error) {
+func readChunk(r io.Reader) ([]byte, error) {
 	// A chunk consists of a length encoded by two bytes and
 	// additional data which is the real value of the item
 	// we reading here from the file.
 
 	b := [2]byte{}
-	if err := readBytes(f, b[:]); err != nil {
+	if _, err := io.ReadFull(r, b[:]); err != nil {
```
Comment (mvo5) [inline x11/xauth.go]: Nice!

File: osutil/cmp.go [hunk 2]
```diff
@@ -63,12 +63,12 @@ func FilesAreEqual(a, b string) bool {
 
 // FileStreamsEqual compares two file streams and returns true if both
 // have the same content.
-func FileStreamsEqual(fa, fb io.Reader) bool {
+func FileStreamsEqual(a, b io.Reader) bool {
```
Comment (pedronis) [inline osutil/cmp.go]: s/File// if I understand the original comments correctly

File: x11/xauth.go [hunk 3]
```diff
@@ -38,61 +38,48 @@ type xauth struct {
...
-	if n != len(data) {
-		return fmt.Errorf("Could not read enough bytes")
-	}
-
-	return nil
-}
-
-func readChunk(f *os.File) ([]byte, error) {
+func readChunk(r io.Reader) ([]byte, error) {
 	// A chunk consists of a length encoded by two bytes and
```
Comment (pedronis) [inline x11/xauth.go]: ... two bytes (so max 64K) ...

**Comment by niemeyer** [comment]
Thank you!

**Pull request merged**